Add mainnet forking to docs (#917)
nomiclabs/buidler: docs/hardhat-network/README.md

- `console.log` is implemented in standard Solidity and then detected in Hardhat Network. This makes its compilation work with any other tools (like Remix, Waffle or Truffle).
- `console.log` calls can run in other networks, like mainnet, kovan, ropsten, etc. They do nothing in those networks, but spend a minimal amount of gas.

## Mainnet forking

The Hardhat Network is empty by default, except for some accounts with an initial balance. But sometimes it's more useful to have a local network that simulates the state of the mainnet. This is what forking is for.

> To fork from the mainnet you need the URL of a node to connect to. For example, using Infura, you can start a local node that forks the mainnet with this command:

Reviewer: ```suggestion
To fork from the mainnet you need the URL of a node to connect to. For example, using Alchemy, you can start a local node that forks the mainnet with this command:
```

```

> npx hardhat node --fork https://mainnet.infura.io/v3/<projectId>

Reviewer: ```suggestion
npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/<key>
```

```

> where you have to replace `<projectId>` with your Infura project id.

Reviewer: ```suggestion
where you have to replace `<key>` with your Alchemy key. You can also use Infura or any other node if you want.
```

After doing this, you can do anything in your node that you can do with a non-forked Hardhat Network: see console logs, get stack traces or use the default accounts to deploy new contracts.

If you want this to be the default behavior, you can do it in your Hardhat config:

```js
networks: {
  hardhat: {
    forking: {

>       url: "https://mainnet.infura.io/v3/<projectId>"

Reviewer: ```suggestion
      url: "https://eth-mainnet.alchemyapi.io/v2/<key>"
```
